docs/source/blocks/examples/customviewvariationsandschemaenhancer.md

```diff
@@ -0,0 +1,274 @@
+import cx from 'classnames';
+import React from 'react';
+
+const View = (props) => {
+  // data holds the values entered in the block edit form
+  // className holds the CSS class names injected to this block by Volto's `styleClassNameExtenders`
+  // style holds the CSS properties injected to this block by Volto's `Block Sytle Wrapper`
+  // variation holds the variation selected in the block editor, and it is an object as defined in the block configuration
+  const { data, className, style, variation } = props;
+
+  const BodyTemplate = variation?.template;
+
+  return (
+    <div
+      className={cx(
+        'block',
+        'block06',
+        `block06-variation-${variation?.id}`,
+        className,
+      )}
+      style={style}
+    >
+      <BodyTemplate data={data} />
+    </div>
+  );
+};
+
+// the `withBlockExtensions` HOC, makes the variation selector available in the block edit form
+// and provides the `variation` property in the props.
+export default withBlockExtensions(View);
+```
+
+## Schema enhancer
+
+We need to configure the schema enhancer function. In this case we will be adding a new field named `color` when using the Variation 2.
+
+Create a file {file}`enhancers.js` in the {file}`BlockSchema` folder with the following content:
+
+```js
+const schemaEnhancerVariation02 = ({ formData, schema, intl }) => {
+  // schema holds the original schema (see the Schema.js file)
+  // so we need to define the new property under `schema.properties`
+  // and push its name to the relevant fieldset, in our case the first one (note the `fieldsets[0]`)
+  schema.properties.color = {
+    title: 'Color',
+  };
+  schema.fieldsets[0].fields.push('color');
+  return schema;
+};
+
+export default schemaEnhancerVariation02;
+
+```
+
+## Variations
+
+Now we need to create one or more variations that will be available for this block.
+
+Create a file {file}`VariationView01.jsx` in the {file}`ExampleBlock06` folder with the following contents:
+
+```jsx
+import React from 'react';
+
+const View = (props) => {
+  const { data } = props;
+  return (
+    <>
+      <h2>Variation View 01</h2>
+      <div>
+        <ul>
+          <li>Title: {data.title}</li>
+          <li>URL: {data.url}</li>
+        </ul>
+      </div>
+    </>
+  );
+};
+
+export default View;
+```
+Create a file {file}`VariationView02.jsx` in the {file}`ExampleBlock06` folder with the following contents:
+
+```jsx
+import React from 'react';
+
+const View = (props) => {
+  const { data } = props;
+  return (
+    <>
+      <h2>Variation View 02</h2>
+      <div>
+        <ul>
+          <li>Title: {data.title}</li>
+          <li>URL: {data.url}</li>
+        </ul>
+      </div>
+    </>
+  );
+};
+
+export default View;
+```
+
+As you can see, in this case the variations are pretty much the same, the only difference is the text that is rendered in the `<h2>` tag. But it can be anything.
```

Comment:
```suggestion
As you can see, the variations are pretty much the same.
The only difference is the text that is rendered in the `<h2>` tag.
But it can be anything.
```